### PR TITLE
Fix announcement emails query

### LINKED ISSUE #1551: Announcement emails still being send to users after a verified email is deleted on profile
Reproduce:

1. Create a new user, turn on announcements by email, add an email address and verify.  Have them join a course.
2. As admin, publish an announcement.  the user above should receive, as expected.
3. As admin, edit the user and delete their email address in their profile.
4. Publish a new announcement.
5. The user STILL receives the new announcement email from the old verified email address, WHY?

Also:
6. As admin, edit the user and untick the "Active" at the very bottom.  Student should now appear in the inactive students list (/profiles/list/inactive/)
7. Publish a new announcement.
5. The user STILL receives the new announcement email despite being inactive, WHY? Is it because they are still registered in a course?  regardless inactive students should be excluded.

### PR DESCRIPTION
**Pull request by yujinyuz**
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/

### What?

Fixes #1551

### Why?

It is reported that users who have their emails removed from their profile still receive email notifications with their previously verified emails.

Also, inactive users still get emails which they shouldn't.

### How?

The first issue can't be reproduced but I added an extra filter that takes the current email address in User.email and includes it in the query.

For the 2nd one, adding in the `is_active=True` to the `CourseStudent.objects.all_users_for_active_semester`

### Testing?

Manually tested and verified that users are not receiving emails

### Review request
@tylerecouture

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved the accuracy of the mailing list by including only verified emails and active users.
- **New Features**
	- Enhanced the announcement emails sending process by optimizing the recipient list generation.

### CODE REVIEW

**Comment by tylerecouture** [comment]
@yujinyuz I'm a bit confused by this comment "users who have disabled receiving announcement emails" <-- where is this reported?  That's not mentioned in the issue.

**Comment by tylerecouture** [comment]
@coderabbitai review

File: src/announcements/tasks.py [hunk 1]
```diff
@@ -44,14 +44,19 @@ def send_announcement_emails(content, root_url, absolute_url):
...
         subject,
         body=text_content,
         to=['contact@bytedeck.com'],
-        bcc=Profile.objects.get_mailing_list(as_emails_list=True, for_announcement_email=True),
+        bcc=profile_emails,
     )
     email_msg.attach_alternative(html_content, "text/html")
     email_msg.send()
+
+    return profile_emails
```
[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/announcements/tasks.py:59]: Do any calls to `send_announcement_emails` use the return value?

**Review by tylerecouture — CHANGES_REQUESTED** [review]
Changes look good, can you ensure they are all covered in testing?

**Comment by tylerecouture** [comment]
> Changes look good, can you ensure they are all covered in testing?

@yujinyuz just missing some test coverage

**Comment by yujinyuz** [comment]
@tylerecouture done adding the testcase

**Pull request merged**